src/packages/auth/src/authentication/resolvers/password/resolver.ts

```diff
@@ -141,18 +110,15 @@ export const createPasswordAuthResolver = <D extends BaseDataEntity>(
 			if (!item.username && !item.password)
 				throw new ValidationError('Update unsuccessful: Nothing to update.');
 
-			let passwordHash = undefined;
-			if (item.password && this.assertPasswordStrength(item.password)) {
-				passwordHash = await hashPassword(item.password);
-			}
-			const credential = await this.provider.updateOne(item.id, {
-				...(item.username ? { username: item.username } : {}),
-				...(passwordHash ? { password: passwordHash } : {}),
+			const entity = await updatePasswordCredential({
+				assertPasswordStrength: this.assertPasswordStrength,
+				provider: this.provider,
+				id: item.id,
+				password: item.password,
+				username: item.username,
+				params,
 			});
 
-			const [entity] = await this.runAfterHooks(HookRegister.AFTER_UPDATE, params, [credential]);
```

Comment:
Out of curiosity, does this run anyway? If so, what triggers it?

Reply:
Might need to talk about this one.

Currently the base resolver runs the hooks, but here we are not using the base resolver so we need to call them directly.

Reply:
This was moved to the utils folder